From c71418d39d5cc0af58548a8885b799efe032c9e3 Mon Sep 17 00:00:00 2001
From: Sughosha <sughosha@disroot.org>
Date: Tue, 16 Dec 2025 08:39:32 +0530
Subject: [PATCH] revert make relocatable

---
 CMakeLists.txt             | 4 ----
 KPimAkonadiConfig.cmake.in | 6 +++---
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c65b2671b..d93ba08f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -399,10 +399,6 @@ configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/KPimAkonadiConfig.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/KPim6AkonadiConfig.cmake"
     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR}
-    PATH_VARS
-        AKONADI_DBUS_INTERFACES_INSTALL_DIR
-        AKONADI_INCLUDE_DIR
-        KF5Akonadi_DATA_DIR
 )
 
 install(
diff --git a/KPimAkonadiConfig.cmake.in b/KPimAkonadiConfig.cmake.in
index 3d1214a5a..2865d1fce 100644
--- a/KPimAkonadiConfig.cmake.in
+++ b/KPimAkonadiConfig.cmake.in
@@ -1,10 +1,10 @@
 @PACKAGE_INIT@
 
-set_and_check(AKONADI_DBUS_INTERFACES_DIR  "@PACKAGE_AKONADI_DBUS_INTERFACES_INSTALL_DIR@")
-set_and_check(AKONADI_INCLUDE_DIR          "@PACKAGE_AKONADI_INCLUDE_DIR@")
+set_and_check(AKONADI_DBUS_INTERFACES_DIR  "@AKONADI_DBUS_INTERFACES_INSTALL_DIR@")
+set_and_check(AKONADI_INCLUDE_DIR          "@AKONADI_INCLUDE_DIR@")
 
 # The directory where akonadi-xml.xsd and kcfg2dbus.xsl are installed
-set(KF5Akonadi_DATA_DIR "@PACKAGE_KF5Akonadi_DATA_DIR@")
+set(KF5Akonadi_DATA_DIR "@KF5Akonadi_DATA_DIR@")
 
 # set the directories
 if(NOT AKONADI_INSTALL_DIR)
-- 
2.52.0

