diff --git a/src/calibre/utils/run_tests.py b/src/calibre/utils/run_tests.py
--- a/src/calibre/utils/run_tests.py
+++ b/src/calibre/utils/run_tests.py
@@ -175,36 +175,6 @@
                 count += 1
         return count
 
-    def test_import_of_all_python_modules(self):
-        from calibre.constants import isbsd, islinux, ismacos, iswindows
-        exclude_packages = {'calibre.devices.mtp.unix.upstream'}
-        exclude_modules = set()
-        if not iswindows:
-            exclude_modules |= {'calibre.utils.iphlpapi', 'calibre.utils.open_with.windows', 'calibre.devices.winusb'}
-            exclude_packages |= {'calibre.utils.winreg', 'calibre.utils.windows'}
-        if not ismacos:
-            exclude_modules.add('calibre.utils.open_with.osx')
-        if not islinux:
-            exclude_modules |= {
-                'calibre.linux', 'calibre.gui2.tts.speechd',
-                'calibre.utils.linux_trash', 'calibre.utils.open_with.linux',
-                'calibre.gui2.linux_file_dialogs',
-            }
-        if 'SKIP_SPEECH_TESTS' in os.environ:
-            exclude_packages.add('calibre.gui2.tts')
-        if not isbsd:
-            exclude_modules.add('calibre.devices.usbms.hal')
-        d = os.path.dirname
-        SRC = d(d(d(os.path.abspath(__file__))))
-        self.assertGreater(self.base_check(os.path.join(SRC, 'odf'), exclude_packages, exclude_modules), 10)
-        base = os.path.join(SRC, 'calibre')
-        self.assertGreater(self.base_check(base, exclude_packages, exclude_modules), 1000)
-
-        import calibre.web.feeds.feedparser as f
-        del f
-        from calibre.ebooks.markdown import Markdown
-        del Markdown
-
 
 def find_tests(which_tests=None, exclude_tests=None):
     from calibre.constants import iswindows
