Add newer language options.

diff --git a/client/langs.go b/tmp/langs.go
index b09c69f..0695958 100644
--- a/client/langs.go
+++ b/tmp/langs.go
@@ -9,6 +9,7 @@ var Langs = map[string]string{
 	"42": "GNU G++11 5.1.0",
 	"50": "GNU G++14 6.4.0",
 	"54": "GNU G++17 7.3.0",
+	"89": "GNU G++20 13.2 (64 bit, winlibs)",
 	"2":  "Microsoft Visual C++ 2010",
 	"59": "Microsoft Visual C++ 2017",
 	"9":  "C# Mono 5.18",
@@ -60,6 +61,7 @@ var LangsExt = map[string]string{
 	"GNU C++11":             "cpp",
 	"GNU C++14":             "cpp",
 	"GNU C++17":             "cpp",
+	"GNU G++20":             "cpp",
 	"MS C++":                "cpp",
 	"MS C++ 2017":           "cpp",
 	"Mono C#":               "cs",
