Resolve a hang in the menu of this game.

Upstream-status: https://github.com/dolphin-emu/dolphin/pull/13285

diff --git a/Data/Sys/GameSettings/GWR.ini b/Data/Sys/GameSettings/GWR.ini
new file mode 100644
index 0000000000..fdbc0c312f
--- /dev/null
+++ b/Data/Sys/GameSettings/GWR.ini
@@ -0,0 +1,5 @@
+# GWR - WAVE RACE / BLUE STORM
+
+[Core]
+# Disable multi-threading to avoid a hang in the game menu.
+CPUThread = False
