From 157524b7cb76c5044a27f4a9e373ee04a9da3c71 Mon Sep 17 00:00:00 2001
From: Ahmad Draidi <a.r.draidi@redscript.org>
Date: Tue, 9 Apr 2024 18:26:52 +0400
Subject: [PATCH] Use arcan from setuid-programs if available

---
 distr/durden | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/distr/durden b/distr/durden
index ab431ce..8672556 100755
--- a/distr/durden
+++ b/distr/durden
@@ -1,5 +1,11 @@
 #!/bin/sh
 
+if [ -n "$(command -v /run/setuid-programs/arcan 2>/dev/null)" ]; then
+	ARCAN_CMD="/run/setuid-programs/arcan"
+else
+	ARCAN_CMD="@ARCAN_STORE_PATH@"
+fi
+
 arcan_base=${HOME}/.arcan
 applname="$(basename $0)"
 distargs=""
@@ -98,11 +104,11 @@ while true; do
 	starttime=$(date +%s)
 
 	if [ -d "${arcan_logpath}" ]; then
-		if arcan ${distargs} -b "$applname" "$applname" "$@" >"${arcan_logpath}/${applname}_${starttime}.log" 2>&1; then
+		if "${ARCAN_CMD}" ${distargs} -b "$applname" "$applname" "$@" >"${arcan_logpath}/${applname}_${starttime}.log" 2>&1; then
 			exit
 		fi
 	else
-		if arcan ${distargs} -b "$applname" "$applname" "$@"; then
+		if "${ARCAN_CMD}" ${distargs} -b "$applname" "$applname" "$@"; then
 			exit
 		fi
 	fi
-- 
2.41.0

