From 24b8d5b63e347bd4cdcf79bf897fc0097fa20fc7 Mon Sep 17 00:00:00 2001
From: Michael Wernthaler <michael.wernthaler@posteo.de>
Date: Thu, 22 Jul 2021 11:27:31 +0200
Subject: [PATCH] Fixed icon for .rs-files

---
 all-the-icons.el | 1 -
 1 file changed, 1 deletion(-)

diff --git a/all-the-icons.el b/all-the-icons.el
index e1f4cb7695..3399906cfb 100644
--- a/all-the-icons.el
+++ b/all-the-icons.el
@@ -194,7 +194,6 @@
     ("r"            all-the-icons-fileicon "R"                :face all-the-icons-lblue)
     ("rd"           all-the-icons-fileicon "R"                :face all-the-icons-lblue)
     ("rdx"          all-the-icons-fileicon "R"                :face all-the-icons-lblue)
-    ("rs"           all-the-icons-fileicon "R"                :face all-the-icons-lblue)
     ("rsx"          all-the-icons-fileicon "R"                :face all-the-icons-lblue)
     ;; There seems to be a a bug with this font icon which does not
     ;; let you propertise it without it reverting to being a lower
