From 233e254bbe44f70ca899a506b3706af489228fda Mon Sep 17 00:00:00 2001
From: aagon <aymeric.agon@hotmail.fr>
Date: Thu, 10 Aug 2023 20:03:52 +0200
Subject: [PATCH] Repair helpful--display-implementations test for emacs 29.1

---
 test/helpful-unit-test.el | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/test/helpful-unit-test.el b/test/helpful-unit-test.el
index 0f5177b..0716b0d 100644
--- a/test/helpful-unit-test.el
+++ b/test/helpful-unit-test.el
@@ -1089,8 +1089,12 @@ find the source code."
   (require 'xref)
   (helpful-function 'xref-location-marker)
   (should (s-contains-p "Implementations" (buffer-string)))
-  (should (s-contains-p "((l xref-file-location))" (buffer-string)))
-  (should (s-contains-p "((l xref-buffer-location))" (buffer-string))))
+  (should (if (version< emacs-version "29.1")
+	      (s-contains-p "((l xref-file-location))" (buffer-string))
+	    (s-contains-p "(xref-location-marker (L xref-file-location))" (buffer-string))))
+  (should (if (version< emacs-version "29.1")
+	      (s-contains-p "((l xref-buffer-location))" (buffer-string))
+	    (s-contains-p "(xref-location-marker (L xref-buffer-location))" (buffer-string)))))
 
 (defun helpful--boring-advice (orig-fn &rest args)
   (apply orig-fn args))
