Submitted upstream: https://github.com/benblazak/ergodox-firmware/pull/100

diff --git a/src/lib/key-functions/public/special.c b/src/lib/key-functions/public/special.c
index 42aba45..6488137 100644
--- a/src/lib/key-functions/public/special.c
+++ b/src/lib/key-functions/public/special.c
@@ -102,9 +102,9 @@ void kbfun_2_keys_capslock_press_release(void) {
 static uint8_t numpad_layer_id;
 
 static inline void numpad_toggle_numlock(void) {
-	_kbfun_press_release(true, KEY_LockingNumLock);
+	_kbfun_press_release(true, KEYPAD_NumLock_Clear);
 	usb_keyboard_send();
-	_kbfun_press_release(false, KEY_LockingNumLock);
+	_kbfun_press_release(false, KEYPAD_NumLock_Clear);
 	usb_keyboard_send();
 }
 
