Workaround https://github.com/fuzzylite/fuzzylite/issues/85

diff --git a/fuzzylite/test/BenchmarkTest.cpp b/fuzzylite/test/BenchmarkTest.cpp
index f868ec29..464d394d 100644
--- a/fuzzylite/test/BenchmarkTest.cpp
+++ b/fuzzylite/test/BenchmarkTest.cpp
@@ -30,7 +30,7 @@ namespace fl {
     }
 
     TEST_CASE("Benchmarks from FLD files", "[benchmark][fld]") {
-        std::string path = "../../examples/";
+        std::string path = "../examples/";
         typedef std::pair<std::string, int > Example;
         std::vector<Example> examples;
         examples.push_back(Example("mamdani/AllTerms", int(1e4)));
