diff -ru a/gcc/cp/cfns.h b/gcc/cp/cfns.h
--- a/gcc/cp/cfns.h	1970-01-01 01:00:01.000000000 +0100
+++ b/gcc/cp/cfns.h	2023-04-23 09:23:28.883815598 +0200
@@ -53,6 +53,9 @@
 static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
 #endif
 const char * libc_name_p (const char *, unsigned int);
 /* maximum key range = 391, duplicates = 0 */
