diff --git a/plugins/usb-protection/gsd-usb-protection-manager.c b/plugins/usb-protection/gsd-usb-protection-manager.c
index 43644408..fef13645 100644
--- a/plugins/usb-protection/gsd-usb-protection-manager.c
+++ b/plugins/usb-protection/gsd-usb-protection-manager.c
@@ -1089,8 +1089,7 @@ usb_protection_proxy_ready (GObject      *source_object,
 
         manager->screensaver_proxy = gnome_settings_bus_get_screen_saver_proxy ();
         if (!manager->screensaver_proxy) {
-                if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
-                        g_warning ("Failed to connect to screensaver service: %s", error->message);
+                g_warning ("Failed to connect to screensaver service");
                 g_clear_object (&manager->usb_protection);
                 return;
         }
