From 17335d49a3c9e866a81abc051420d73a59d669bf Mon Sep 17 00:00:00 2001
Message-Id: <17335d49a3c9e866a81abc051420d73a59d669bf.1678745341.git.dev@jpoiret.xyz>
From: Josselin Poiret <dev@jpoiret.xyz>
Date: Sun, 28 Aug 2022 01:17:34 +0200
Subject: [PATCH] Add missing const_mach_port_name_array_t type

From: Samuel Thibault <samuel.thibault@ens-lyon.org>

---
 include/mach/port.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/mach/port.h b/include/mach/port.h
index 3c226f6c..d25eb0b4 100644
--- a/include/mach/port.h
+++ b/include/mach/port.h
@@ -43,6 +43,7 @@
  */
 typedef unsigned int mach_port_name_t;
 typedef mach_port_name_t *mach_port_name_array_t;
+typedef const mach_port_name_t *const_mach_port_name_array_t;
 
 /*
  * A port is represented

base-commit: 3e1702a65fb3caf50c8d09a1b383b0056a5efc82
prerequisite-patch-id: 4688654277aef235ccd1797a72dce27d52b04616
prerequisite-patch-id: 8f1e7dd6429c15cc8ed1d411dba0a75b6e745236
prerequisite-patch-id: 605553ee086473ad9bc24e59c91d49a12dbf8631
-- 
2.39.1

