From 8f4942f0612b6cc6af56fc90146afcccfe67d85f Mon Sep 17 00:00:00 2001
From: Grigory Shepelev <shegeley@gmail.com>
Date: Sat, 25 May 2024 10:32:28 +0300
Subject: [PATCH] srfi/125.sld: conform "rename" clause to r6rs

---
 srfi/125.sld | 25 ++++++++++++-------------
 1 file changed, 12 insertions(+), 13 deletions(-)

diff --git a/srfi/125.sld b/srfi/125.sld
index 08d9ac1..467c6a8 100644
--- a/srfi/125.sld
+++ b/srfi/125.sld
@@ -53,19 +53,18 @@
 
    ;; The following procedures are deprecated by SRFI 125:
 
-   (rename deprecated:hash                     hash)
-   (rename deprecated:string-hash              string-hash)
-   (rename deprecated:string-ci-hash           string-ci-hash)
-   (rename deprecated:hash-by-identity         hash-by-identity)
-
-   (rename deprecated:hash-table-equivalence-function
-                                               hash-table-equivalence-function)
-   (rename deprecated:hash-table-hash-function hash-table-hash-function)
-   (rename deprecated:hash-table-exists?       hash-table-exists?)
-   (rename deprecated:hash-table-walk          hash-table-walk)
-   (rename deprecated:hash-table-merge!        hash-table-merge!)
-
-   )
+   (rename
+    (deprecated:hash                            hash)
+    (deprecated:string-hash                     string-hash)
+    (deprecated:string-ci-hash                  string-ci-hash)
+    (deprecated:hash-by-identity                hash-by-identity)
+    (deprecated:string-ci-hash                  string-ci-hash)
+    (deprecated:hash-by-identity                hash-by-identity)
+    (deprecated:hash-table-equivalence-function hash-table-equivalence-function)
+    (deprecated:hash-table-hash-function        hash-table-hash-function)
+    (deprecated:hash-table-exists?              hash-table-exists?)
+    (deprecated:hash-table-walk                 hash-table-walk)
+    (deprecated:hash-table-merge!               hash-table-merge!)))
 
   (import (scheme base)
           (scheme write) ; for warnings about deprecated features
-- 
2.49.0

