From: Boyuan Yang <byang@debian.org>
Date: Tue, 30 Jul 2024 13:59:56 -0400
Subject: gcc14 fix

Bug-Debian: https://bugs.debian.org/1074929
---
 libebutils/puts_eucjp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libebutils/puts_eucjp.c b/libebutils/puts_eucjp.c
index 678b296..fb21920 100644
--- a/libebutils/puts_eucjp.c
+++ b/libebutils/puts_eucjp.c
@@ -59,7 +59,7 @@ fputs_eucjp_to_locale(const char *string, FILE *stream)
     char *buffer = NULL;
     size_t buffer_size;
     iconv_t cd = (iconv_t)-1;
-    const char *in_p;
+    char *in_p;
     char *out_p;
     size_t in_left;
     size_t out_left;
@@ -89,7 +89,7 @@ fputs_eucjp_to_locale(const char *string, FILE *stream)
 	if (buffer == NULL)
 	    goto failed;
 
-	in_p = string;
+	in_p = (char *) string;
 	in_left = string_length + 1;
 	out_p = buffer;
 	out_left = buffer_size;

