Upstream-status: <https://review.jami.net/c/jami-daemon/+/32708>

diff --git a/configure.ac b/configure.ac
index 055151670..f6e4ae978 100644
--- a/configure.ac
+++ b/configure.ac
@@ -590,6 +590,8 @@ PKG_CHECK_MODULES([SECP256K1], [libsecp256k1 >= 0.1],,
 PKG_CHECK_MODULES([FMT], [fmt >= 7.0],,
   AC_MSG_FAILURE([Missing {fmt}]))
 
+PKG_CHECK_MODULES([SIMDUTF], [simdutf])
+
 # SHM
 dnl check for shm
 AS_IF([test "x$with_dbus" == xyes],
diff --git a/meson.build b/meson.build
index c0f98c7f8..3ad0bd0ba 100644
--- a/meson.build
+++ b/meson.build
@@ -32,6 +32,7 @@ deplibavutil      = dependency('libavutil', version: '>= 55.75.100')
 depfmt            = dependency('fmt', version: '>= 5.3')
 depyamlcpp        = dependency('yaml-cpp', version: '>= 0.5.1')
 depjsoncpp        = dependency('jsoncpp', version: '>= 1.6.5')
+depsimdutf        = dependency('simdutf')
 depzlib           = dependency('zlib')
 
 if get_option('interfaces').contains('dbus')
diff --git a/src/Makefile.am b/src/Makefile.am
index 8d874223e..cee546126 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -52,6 +52,7 @@ libjami_la_LIBADD += \
 	$(FMT_LIBS) \
 	$(LIBSSL_LIBS) \
 	$(LIBCRYPTO_LIBS) \
+	$(SIMDUTF_LIBS) \
 	$(ZLIB_LIBS)
 
 # XXX: It seems this should be the default value when otherwise left
diff --git a/src/meson.build b/src/meson.build
index 22d026abc..bfc809cba 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -140,6 +140,7 @@ libjami_dependencies = [
     depfmt,
     depyamlcpp,
     depjsoncpp,
+    depsimdutf,
     depzlib,
     depopenssl
 ]
