Don't escape forward slashes to prevent a warning from grep 3.8.

diff --git a/tests/link-order.at b/tests/link-order.at
--- a/tests/link-order.at
+++ b/tests/link-order.at
@@ -99,12 +99,12 @@ aix* | interix*) ;;  # These systems have different path syntax
   case $hardcode_direct$hardcode_direct_absolute in
   yesno)
     AT_CHECK([if $EGREP relinking stderr; then
-         $EGREP " .*\/new\/lib/libb$shared_ext .*\/old\/lib/libcee$shared_ext" stdout
+         $EGREP " .*/new/lib/libb$shared_ext .*/old/lib/libcee$shared_ext" stdout
        else :; fi], [0], [ignore], [], [echo "wrong link order"])
     ;;
   *)
     AT_CHECK([if $EGREP relinking stderr; then
-         $EGREP " -L.*\/new\/lib -lb -L.*\/old\/lib -lcee" stdout
+         $EGREP " -L.*/new/lib -lb -L.*/old/lib -lcee" stdout
        else :; fi], [0], [ignore], [], [echo "wrong link order"])
     ;;
   esac
diff --git a/tests/testsuite b/tests/testsuite
--- a/tests/testsuite
+++ b/tests/testsuite
@@ -20611,11 +20611,11 @@ aix* | interix*) ;;  # These systems have different path syntax
   yesno)
     { set +x
 $as_echo "$at_srcdir/link-order.at:101: if \$EGREP relinking stderr; then
-         \$EGREP \" .*\\/new\\/lib/libb\$shared_ext .*\\/old\\/lib/libcee\$shared_ext\" stdout
+         \$EGREP \" .*/new/lib/libb\$shared_ext .*/old/lib/libcee\$shared_ext\" stdout
        else :; fi"
 at_fn_check_prepare_notrace 'an embedded newline' "link-order.at:101"
 ( $at_check_trace; if $EGREP relinking stderr; then
-         $EGREP " .*\/new\/lib/libb$shared_ext .*\/old\/lib/libcee$shared_ext" stdout
+         $EGREP " .*/new/lib/libb$shared_ext .*/old/lib/libcee$shared_ext" stdout
        else :; fi
 ) >>"$at_stdout" 2>>"$at_stderr" 5>&-
 at_status=$? at_failed=false
@@ -20633,11 +20633,11 @@ $at_traceon; }
   *)
     { set +x
 $as_echo "$at_srcdir/link-order.at:106: if \$EGREP relinking stderr; then
-         \$EGREP \" -L.*\\/new\\/lib -lb -L.*\\/old\\/lib -lcee\" stdout
+         \$EGREP \" -L.*/new\\/lib -lb -L.*/old/lib -lcee\" stdout
        else :; fi"
 at_fn_check_prepare_notrace 'an embedded newline' "link-order.at:106"
 ( $at_check_trace; if $EGREP relinking stderr; then
-         $EGREP " -L.*\/new\/lib -lb -L.*\/old\/lib -lcee" stdout
+         $EGREP " -L.*/new/lib -lb -L.*/old/lib -lcee" stdout
        else :; fi
 ) >>"$at_stdout" 2>>"$at_stderr" 5>&-
 at_status=$? at_failed=false
