diff --git a/data/mono.pc.in b/data/mono.pc.in
index 6da0960db2d..d43bb187218 100644
--- a/data/mono.pc.in
+++ b/data/mono.pc.in
@@ -7,6 +7,6 @@ sysconfdir=@sysconfdir@
 Name: Mono
 Description: Mono Runtime
 Version: @VERSION@
-Requires: glib-2.0 gthread-2.0
+Requires: glib-2.0 gthread-2.0 bdw-gc
 Libs: -L${libdir} @export_ldflags@ -lmono @libmono_ldflags@
 Cflags: -I${includedir} @libmono_cflags@
diff --git a/mono-uninstalled.pc.in b/mono-uninstalled.pc.in
index 7fa3f12dc91..2a0734362fd 100644
--- a/mono-uninstalled.pc.in
+++ b/mono-uninstalled.pc.in
@@ -1,6 +1,6 @@
 Name: Mono
 Description: Mono Runtime
 Version: @VERSION@
-Requires: glib-2.0 gthread-2.0
+Requires: glib-2.0 gthread-2.0 bdw-gc
 Libs: -L@mono_build_root@/mono/mini/.libs @export_ldflags@ -lmono @libmono_ldflags@
 Cflags: -I@abs_top_srcdir@ -I@abs_top_srcdir@/mono @libmono_cflags@
diff --git a/mono/metadata/Makefile.am b/mono/metadata/Makefile.am
index 83f8532369b..3fca7fc13b9 100644
--- a/mono/metadata/Makefile.am
+++ b/mono/metadata/Makefile.am
@@ -171,7 +171,6 @@ libmonoruntimeinclude_HEADERS = \
 	object.h	\
 	exception.h	\
 	profiler.h	\
-	appdomain.h	\
 	mono-config.h	\
 	debug-helpers.h	\
 	mempool.h
diff --git a/mono/mini/driver.c b/mono/mini/driver.c
index 48ca2d96899..9fb3512200b 100644
--- a/mono/mini/driver.c
+++ b/mono/mini/driver.c
@@ -1236,6 +1236,7 @@ mono_main (int argc, char* argv[])
 #endif
 	if (!g_thread_supported ())
 		g_thread_init (NULL);
+        GC_allow_register_threads();
 
 	if (mono_running_on_valgrind () && getenv ("MONO_VALGRIND_LEAK_CHECK")) {
 		GMemVTable mem_vtable;
diff --git a/runtime/Makefile.am b/runtime/Makefile.am
index 0e876e2a491..b9cd0a99d9e 100644
--- a/runtime/Makefile.am
+++ b/runtime/Makefile.am
@@ -1,6 +1,3 @@
-# hack to prevent 'check' from depending on 'all'
-AUTOMAKE_OPTIONS = cygnus
-
 tmpinst = _tmpinst
 
 noinst_SCRIPTS = mono-wrapper monodis-wrapper
