Upstream-status: <https://github.com/savoirfairelinux/opendht/pull/821>

diff --git a/meson.build b/meson.build
index d3c4a500..8d1ed563 100644
--- a/meson.build
+++ b/meson.build
@@ -116,6 +116,13 @@ opendht = library(
     install: true,
 )
 
+# Install library headers.
+install_subdir(
+    'include',
+    install_dir: get_option('includedir'),
+    strip_directory: true
+)
+
 conf_data.set('VERSION', meson.project_version())
 conf_data.set('prefix', get_option('prefix'))
 conf_data.set('exec_prefix', get_option('prefix'))
@@ -145,6 +152,11 @@ if get_option('c').enabled()
         install: true,
     )
 
+    install_headers(
+        'c/opendht_c.h',
+        subdir: 'opendht'
+    )
+
     configure_file(
         input: 'opendht-c.pc.in',
         output: 'opendht-c.pc',
