Upstream-status: <https://github.com/NetworkConfiguration/openresolv/pull/51>

diff --git a/resolvconf.in b/resolvconf.in
index 2f55a9d..c020413 100644
--- a/resolvconf.in
+++ b/resolvconf.in
@@ -475,6 +475,13 @@ detect_init()
 			fi'
 	elif [ -d /etc/dinit.d ] && command -v dinitctl >/dev/null 2>&1; then
 		RESTARTCMD='dinitctl --quiet restart --ignore-unstarted $1'
+	elif [ -x /run/current-system/profile/bin/herd ] >/dev/null 2>&1; then
+		# Guix System
+		RESTARTCMD='
+			if /run/current-system/profile/bin/herd eval root "(unless (lookup-running '\''$1) (error))" >/dev/null 2>&1
+			then
+				/run/current-system/profile/bin/herd restart $1
+			fi'
 	else
 		for x in /etc/init.d/rc.d /etc/rc.d /etc/init.d; do
 			[ -d $x ] || continue
