Upstream-status: Not presented upstream.

--- pciutils-3.8.0/lib/hurd.c.orig	2024-11-07 15:09:20.728973822 +0100
+++ pciutils-3.8.0/lib/hurd.c	2024-11-07 15:05:08.524682813 +0100
@@ -213,7 +213,7 @@
 hurd_read(struct pci_dev *d, int pos, byte * buf, int len)
 {
   int err;
-  size_t nread;
+  mach_msg_type_number_t nread;
   char *data;
   mach_port_t device_port = device_port_lookup(d);
 
@@ -266,7 +266,7 @@
   mach_port_t device_port = device_port_lookup(d);
   struct pci_bar regions[6];
   char *buf = (char *) &regions;
-  size_t size = sizeof(regions);
+  mach_msg_type_number_t size = sizeof(regions);
 
   int err = pci_get_dev_regions(device_port, &buf, &size);
   if (err)
@@ -305,7 +305,7 @@
   struct pci_xrom_bar rom;
   mach_port_t device_port = device_port_lookup(d);
   char *buf = (char *) &rom;
-  size_t size = sizeof(rom);
+  mach_msg_type_number_t size = sizeof(rom);
 
   int err = pci_get_dev_rom(device_port, &buf, &size);
   if (err)
