diff --git a/plugins/tb/transferbench/CMakeLists.txt b/plugins/tb/transferbench/CMakeLists.txt
index 98e4bc4..8e4588e 100644
--- a/plugins/tb/transferbench/CMakeLists.txt
+++ b/plugins/tb/transferbench/CMakeLists.txt
@@ -153,7 +153,7 @@ endif()
 
 ## Check for hsa support
 find_library(HSA_LIBRARY hsa-runtime64 PATHS ${ROCM_PATH} ${ROCM_PATH}/lib)
-find_path(HSA_INCLUDE_DIR hsa.h PATHS ${ROCM_PATH}/include ${ROCM_PATH}/include/hsa)
+find_path(HSA_INCLUDE_DIR has/hsa.h PATHS ${ROCM_PATH}/include ${ROCM_PATH}/include/hsa)
 if(HSA_LIBRARY AND HSA_INCLUDE_DIR)
     add_library(hsa-runtime64 SHARED IMPORTED)
     set_target_properties(hsa-runtime64 PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${HSA_INCLUDE_DIR}" IMPORTED_LOCATION "${HSA_LIBRARY}" INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${HSA_INCLUDE_DIR}")
-- 
2.51.0

