This patch fixes the unknown version comparison operator

diff --git a/librtosc-cpp.pc.cmake b/librtosc-cpp.pc.cmake
index 0d8fda2..edbcb3a 100644
--- a/librtosc-cpp.pc.cmake
+++ b/librtosc-cpp.pc.cmake
@@ -9,6 +9,6 @@ includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 Name: rtosc_cpp
 Description: rtosc_cpp - a realtime safe open sound control serialization and dispatch system for C++
 Version: @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
-Requires: librtosc == @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
+Requires: librtosc = @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
 Libs: -L${libdir} -lrtosc -lrtosc-cpp
 Cflags: -I${includedir}