--- openssl-sys-0.9.84/Cargo.toml.orig	2023-04-03 09:10:11.979197979 -0400
+++ openssl-sys-0.9.84/Cargo.toml	2023-04-03 12:07:30.285315609 -0400
@@ -31,10 +31,6 @@
 [package.metadata.pkg-config]
 openssl = "1.0.1"
 
-[dependencies.bssl-sys]
-version = "0.1.0"
-optional = true
-
 [dependencies.libc]
 version = "0.2"
 
@@ -46,16 +42,12 @@
 [build-dependencies.cc]
 version = "1.0"
 
-[build-dependencies.openssl-src]
-version = "111"
-optional = true
-
 [build-dependencies.pkg-config]
 version = "0.3.9"
 
 [features]
-unstable_boringssl = ["bssl-sys"]
-vendored = ["openssl-src"]
+unstable_boringssl = []
+vendored = []
 
 [target."cfg(target_env = \"msvc\")".build-dependencies.vcpkg]
 version = "0.2.8"
