From b7750fa99c31c5f28ac2053fa154bef4038b5cc0 Mon Sep 17 00:00:00 2001
From: dan <i@dan.games>
Date: Tue, 2 Jul 2024 19:37:38 +0800
Subject: [PATCH] tests: remove portal tests.

---
 tests/meson.build | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/tests/meson.build b/tests/meson.build
index a2dafee..bcd87be 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -181,18 +181,6 @@ portal_tests = [
 
 test_env = env_tests
 test_env.set('XDG_CURRENT_DESKTOP', 'test')
-foreach p : portal_tests
-    test(
-      'test-portals-@0@'.format(p),
-      test_portals,
-      args: ['--verbose', '--keep-going', '--tap', '-p', '/portal/@0@'.format(p)],
-      depends: [test_backends, test_portals],
-      env: test_env,
-      is_parallel: false,
-      protocol: test_protocol,
-      suite: 'portals',
-    )
-endforeach
 
 # Split the portal tests into one test per portal, this makes debugging a lot
 # easier.
@@ -204,18 +192,6 @@ portal_limited = [
 
 limited_env = env_tests
 limited_env.set('XDG_CURRENT_DESKTOP', 'limited')
-foreach p : portal_limited
-    test(
-      'limited-portals-@0@'.format(p),
-      limited_portals,
-      args: ['--verbose', '--keep-going', '--tap', '-p', '/limited/@0@'.format(p)],
-      depends: [test_backends, limited_portals],
-      env: limited_env,
-      is_parallel: false,
-      protocol: test_protocol,
-      suite: 'portals',
-    )
-endforeach
 
 if enable_installed_tests
   install_data(

base-commit: 11c8a96b147aeae70e3f770313f93b367d53fedd
-- 
2.41.0

